### DISCUSSION TITLE
Global Filter

### DISCUSSION THREAD

**Discussion by mustrockux** [vmware/clarity#5759] [Ideas]
_This contribution is from the designers on Project Iris, an application modernization product in developement_

## Summary

_We are proposing a global filter for more advance filtering of a more complex elements on a page with mutiple dimensions/categories other than a grid. This is an enhancement of the in column filter and sort capability and a more complex function of sorting elements that are not visible to the user on the page. Clarity does not have this abaility and as we extend the card design, we may need to allow users more complex filtering and sorting, or "slicing and dicing:" functions of traditional grids as in excel or tableau. Currently, the Clarity does not allow the sorting or filtering of dimensions other than metadata exposed in the grid in a column._

## Use case
1. Several cards are on a page and a user needs to sort them by a dimension or category that is visible on the card
2. Many cards (over 50) are on a page and user needs to filter their view by a category or dimension
3. A grid has many rows and a user needs to filter by more than one category and one time 
4. A user wants to see in a static fashion the filters that were engaged in grid

## Implementation plan
I am not sure how to implement it in the code and would love some help on how to do it.

**Comment by mrmokwa** [comment]
I'd create a singleton service with a behavior subject that would hold the value of the global filters.
Then when the datagrid.refresh() is dispatch, you could merge the information of the event with the values of the subject.

**Comment by Jinnie** [comment]
A common approach I've seen would be to have a combobox (with multiselection), or similar component, then select "tags"
 which will be used as base for the filtering. A tag could even be a key-value pair if you want to filter on specific fields values. Valid filtering values are provided as suggestions in the combobox dropdown.
